### ISSUE TITLE
IDE0200 mis-fires, 'fixes' code to be defective

### ISSUE THREAD

**Issue by AArnott** [dotnet/roslyn#69094]
**Version Used**: VS 17.8 Preview 1 (33909.54.main)

**Steps to Reproduce**:

[See the sharplab repro](https://sharplab.io/#v2:C4LgTgrgdgNAJiA1AHwAICYCMBYAUKgBgAJVMBWAbjz1QDYBhAewFsAHAGwFNgBLRqAMqMIYAMacAPACNGjdgD4iANwCG7CJwHciAXiJROAdwAUASiq4AEp3atOYIgAsbdh3oMmluxavWbuAHRawABKnADOEOzAxkqm5njOtvZB3MbAkJwJuL4awbr6RmYWSa6pMQBmauFZFjToRNbJYHgA3nhEnaxgPKrAnERgnCpw/OwAniSY6NKyCkQqorz8Fp1EPFD9YFBqjS72xqQzMnKKi8tQph2d7bhrncCOPOEB53xQBW8r10QAvtR3TobLY7dgkAAsRGCxhOYNyWW8CyW71iag02X+uCAA=)

Try accepting the suggested code-fix on the lambda applies this change:
```diff
-Helper helper = new(v => valueSet.SetResult(v));
+Helper helper = new(valueSet.SetResult);
```

But as you can see in the executed code that sharplab produces, this breaks the app.

In short, in the case below:

```cs
TaskCompletionSource<bool> valueSet = new();
Helper helper = new(v => valueSet.SetResult(v));
```

Reducing the lambda to just a method group as IDE0200 would like to see is _not_ the same thing. If `valueSet` is reassigned, the method group syntax will still point to the original value, whereas the lambda syntax will use the new value.

**Diagnostic Id**: IDE0200

**Expected Behavior**:

IDE0200 notices that `valueSet` is reassigned and does _not_ suggest removing the lambda syntax since that would change behavior and introduce the bug.

**Actual Behavior**:

IDE0200 suggests removing the lambda is equivalent, but taking that action causes a runtime failure.
